### ISSUE TITLE
Deprecation warning appears to be unavoidable

### ISSUE THREAD

**Issue by nevinera** [fastly/fastly-ruby#173]
I'm getting this warning locally: https://github.com/fastly/fastly-ruby/blob/master/lib/fastly/client.rb#L28-L30

> DEPRECATION WARNING: Username/password authentication is deprecated and will not be available starting September 2020; please migrate to API tokens as soon as possible.

It appears that that warning is produced on any instantiation of `Fastly::Client`, which is performed on any instantiation of `Fastly` - am I missing something? I don't see any way to use this gem without producing that warning. Building an instance as described in [the docs](https://github.com/fastly/fastly-ruby#examples) results in that message, for example:

```
2.7.2 :001 > require 'fastly'
 => true
2.7.2 :002 > fastly = Fastly.new(api_key: "fake-key")
DEPRECATION WARNING: Username/password authentication is deprecated
      and will not be available starting September 2020;
      please migrate to API tokens as soon as possible.
 => #<Fastly:0x00007fb06ea30ea0 @client=#<Fastly::Client:0x00007fb06ea30e50 @api_key="fake-key", @base_url="https://api....
2.7.2 :003 >
```

We are *actually* instantiating like this in production: `Fastly.new({ api_key: ENV.fetch("FASTLY_VIDEO_SERVICE_PURGING_TOKEN")` - given that it didn't stop working _late last year_, can I assume that that is the appropriate way to use the gem, and that the deprecation warning is in error?

**Comment by conniechu929** [comment]
If you have previously used an `api_key` to log in, please disregard the error. Sorry about the confusion. The deprecation  warning is within the `Client` so that's why you see that every time you instantiate a new client. I will go and make a ticket to remove this deprecation warning now that the deprecation has _actually_ happened. Sorry again if there was a confusion.

**Comment by nevinera** [comment]
Ah, that makes sense. thanks!

**Comment by Integralist** [comment]
Closing this issue now we've published v4.0.0 https://rubygems.org/gems/fastly
